Implementation of alias editor and usage in feature grid

offtherailz opened the pull request:
## Description

The current PR provides

- A small refactor of the layer setting panel (for clean up organization)
- An new tab in layer settings, with Fields configuration. It provides the possibility to edit the aliases (optionally localized) #9127 
- The implementation of usage of Alias for the feature grid (part of #9128 )

@allyoucanmap : For the moment, the 3 features above are provided in 3 different commits, for an easier review, hope it helps to have a clear overview of the changes applied

The other implementations for #9099 (Visual Style Editor, Query Panel, Filter Layer tool, Identify, Widgets) are still pending. 
We can evaluate if merge this and then the other implementations as separate pull requests or continue the work on this branch and then merge all the work together.

Anyway I can suggest to start reviewing this work from now, so we can have less files to review at a single time. 

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [x] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

<!-- add here the ReadTheDocs link (if needed) -->

## Issue

**What is the current behavior?**
<!-- You can also link to an existing issue here -->
#<issue>

**What is the new behavior?**
<!-- Describe here the new behaviour based on your changes -->

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications -->

## Other useful information

offtherailz commented:
A suggestion from @allyoucanmap to check, is to remove the shadow near the tabs, to have a better visual effect.

![image](https://user-images.githubusercontent.com/1279510/234508763-0adca285-4cee-4610-9f33-f167a9a0651e.png)

I'll try to give you the result and check if it works also for the other tabs ( it was already there for some reason I suppose)

allyoucanmap commented on web/client/components/TOC/fragments/LayerFields/index.jsx:
```diff
@@ -0,0 +1,106 @@
+        setLoading(true);
+        loadFields(layer, merge)
+            .then((newFields) => {
+                setLoading(false);
+                setFields(newFields);
+            })
+            .catch((e) => {
+                setLoading(false);
+                setError(e);
```
I think we should check if the component is still mounted to avoid to trigger internal state while the component is not mounted anymore

allyoucanmap reviewed (commented):
The code seems good, just some things to check:

- alias is not currently save in the exported or saved map
- should we allow to edit the name of the geometry field?
- about the style here the result without shadows:

![image](https://user-images.githubusercontent.com/19175505/234866046-763d7dff-3af5-4912-b180-8207e405db5c.png)

tdipisa commented:
> The code seems good, just some things to check:
> 
> * alias is not currently save in the exported or saved map

@offtherailz correct me if I'm wrong but this is part of a dedicated issue mentioned in the task list of [#9099](https://github.com/geosolutions-it/MapStore2/issues/9099#issue-1668414178) that still need to be defined

> * should we allow to edit the name of the geometry field?

I think we can omit the geometry field which is never used in the UI to display its value (eg. in Attribute Table or Identify or other tools)

offtherailz commented:
- Style

@allyoucanmap can you provide the style changes you used or apply them direcly on the branch?

- Saving

```
correct me if I'm wrong but this is part of a dedicated issue mentioned in the task list of https://github.com/geosolutions-it/MapStore2/issues/9099#issue-1668414178 that still need to be defined
```
Yes you are right, it is part of a dedicated task in the task list, anyway it makes sense to include it in the current PR for consistency. So it is ok. 

```
hide the gemetry
```
 ok i can exclude it from the UI

offtherailz commented:
@allyoucanmap I fixed the parts you asked. 
Notes: 
- I avoided to remove the shadow from the top it is applied on `ms-header`, and when I tried to customize this component of TOCItemsSettings I had this result.

![image](https://user-images.githubusercontent.com/1279510/234925550-ec8fd8e1-3197-4e7c-8502-fc39f455ed09.png)

- I added a light border to better separate header and footer of the scrollable content from the content itself.

offtherailz commented:
@ElenaGallo, could you please test this on  DEV ? Thank you

ElenaGallo commented:
Hi @offtherailz, I noticed that:

- the shadow under the settings tabs is still present
![shadow](https://github.com/geosolutions-it/MapStore2/assets/56537133/561009b4-4afd-45ea-b892-d08c8ef8a641)

- when Alias is removed, in the attribute table it is not replaced with the name of the attribute but the empty space is left
(only by using the 'Clear customization' button does the attribute name become visible again in the attribute table)
https://github.com/geosolutions-it/MapStore2/assets/56537133/64b23c0b-42b7-4093-bc5d-14231793db77
